src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -51,6 +55,21 @@ public bool IsExperimentEnabled(string experimentName)
             ThisCanBeCalledOnAnyThread();
             if (_isCachedFlightEnabledInfo != null)
             {
+                try
+                {
+                    var enabled = _featureFlags.IsFeatureEnabled(experimentName, defaultValue: false);
+                    if (enabled)
+                    {
+                        return enabled;
+                    }
+
+                }
+                catch
+                {
+                    // featureFlags can throw if given name is in incorrect format which can happen for us
+                    // since we use this for experimentation service as well
```

Comment:
can we catch a particular exception?  this is literally a catch-all, which means *any* problem whatever will be masked here.

Reply:
catch and do what? I dont want us to crash. we will see whether even NFW is useful.

Reply:
also, apparently, we were fine doing catch all for existing experiment API so, I don't see what makes this one different.

Reply:
> catch and do what? I dont want us to crash. 

That's not a good reason to just 'catch all'.  Catch-alls are exceptionally (no pun intended) broad, and shouldn't be used in our codebase.  If there's an exception we know this API will produce, and we can't avoid that, we should just catch that specific exception

> also, apparently, we were fine doing catch all for existing experiment API so, I don't see what makes this one different.

Other code being poorly written isn't a reason to perpetuate bad coding practices.  'catch all's are really bad as they can mask significant bugs and issues.  You can read lots more about this in many resources out there (including several published by MS i believe).  I'm fairly certain though we've gone over this in the Roslyn time a half dozen times or so, so this is not new information.

Also, I'm happy if you want to fix up those other catches at the same time as the work you're currently doing here.  The presence of existing broken windows is definitely not a reason or excuse to add more broken ones in the present.

Reply:
I also said I don't agree so, I am not going to keep writing here anymore. the code was there for years and we didn't have any issue with it, I don't see any reason to change it now. unless you can prove that there is a benefit to we can see from us starting to behave differently I am going to leave it as it is.

probably better way will be you finding someone asking same thing.

Reply:
I agree that the discussion about using a catch all or not should be moved to a separate issue as this PR is following an existing pattern used for similar functionality in this service. Unless `IVsFeatureFlags.IsFeatureEnabled` has publically documented set of exceptions which cannot change in future and we can just catch them.

Reply:
>the code was there for years and we didn't have any issue with it, I don't see any reason to change it now. 

Because it's not good code.  It's not written responsibility and it harms the health of the codebase.  As the stewards of this code we should be striving to write things with proper patterns, and that includes both addressing problems as we see them and *not* continuing to introduce the same negative patterns into the codebase.

> unless you can prove that there is a benefit to we can see from us starting to behave differently I am going to leave it as it is.

Yes.  That's an easy proof to make.  Currently, the code catches all exceptions.  This means that it will catch critical product bugs that break functionality.  By doing this you both mask those bugs and prevent MS from fixing them.  Furthermore, this means users can now experience indeterminate broken behavior without any way to figure out why.

Finally, this may affect our actual ability to do feature tests as we may fail in these lower layers and then totally swallow things.

I'm actually baffled you would make this argument.  THe problems with 'catch all' have been known for decades.  We've discussed it on the team at least a dozen times.  You and me have *personally* discussed on a few occasions.  We've seen what happens here with catch-alls.  And, frankly, i'm a bit surprised they're just not all-out banned in the Roslyn codebase.